core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -898,15 +897,8 @@ public SegmentFile getSegmentFile() {
    * @return
    */
   public List<CarbonFile> getIndexCarbonFiles() {
-    Map<String, String> indexFiles = getIndexFiles();
-    Set<String> files = new HashSet<>();
-    for (Map.Entry<String, String> entry: indexFiles.entrySet()) {
-      Path path = new Path(entry.getKey());
-      files.add(entry.getKey());
-      if (entry.getValue() != null) {
-        files.add(new Path(path.getParent(), entry.getValue()).toString());
-      }
-    }
+    List<String> indexFiles = getIndexFiles();
```

Comment:
Can you return Set<String> from getIndexFiles itself

Reply:
done